src/util/expr.cpp

```diff
@@ -245,7 +245,7 @@ bool exprt::is_zero() const
     {
       rationalt rat_value;
       if(to_rational(*this, rat_value))
-        assert(false);
+        CHECK_RETURN(false);
```

Comment:
It would actually be even more convenient if we could write `CHECK_RETURN(to_rational(*this, rat_value));`. However, we must guarantee that the implementation of CHECK_RETURN always evaluates its argument.